app/scripts/controllers/centers/CreateCenterController.js

```diff
@@ -21,6 +21,11 @@
             scope.getGroups = function() {
                 resourceFactory.groupResource.getAllGroups({officeId: scope.formData.officeId }, function (data) {
                     scope.groups = data;
+                    for (var i = 0; i<scope.groups.length; i++) {
+                        if (scope.groups[i].centerName) {
+                            scope.groups.splice(i, 1);
+                        }
+                    }
```

Comment:
There is an issue with this approach. As you are iterating over the array using a for loop and at the same time modifying it with `splice`, there might be unexpected results in some cases as elements are removed from the array and its length is modified during iteration. Try using `filter` method instead. Also, the id parameter of the center can be used instead of name for filtering.